### PR TITLE
Fixing macos CI

### PR DESCRIPTION
**Pull request by sebastienros**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,26 +1,25 @@
 language: csharp
-sudo: false
+sudo: required
 dist: trusty
-dotnet: 1.0.0
+mono: none
+dotnet: 1.0.1
```
Comment (alexbocharov) [inline .travis.yml]: Why not `dotnet: 1.0.3`?

File: .travis.yml [hunk 2]
```diff
@@ -1,26 +1,25 @@
...
+env:
+  global:
+    - DOTNET_SKIP_FIRST_TIME_EXPERIENCE: true
+    - DOTNET_CLI_TELEMETRY_OPTOUT: 1   
+    - TEMP: "$TRAVIS_BUILD_DIR/.temp"
 addons:
```
Comment (alexbocharov) [inline .travis.yml]: Can be removed

Comment (sebastienros) [inline .travis.yml]: Then it fails ... I twill try again once I get greens. linux is green right now, waiting for macos.

File: .travis.yml (hunk 1 above)
Comment (sebastienros) [inline .travis.yml]: I don't really know what travis supports.

**Comment by sebastienros** [comment]
Good, I got all greens, appveyor and travis, including unit tests.
@alexbocharov Now I will remove the things you suggested one by one.

File: .travis.yml (hunk 1 above)
Comment (alexbocharov) [inline .travis.yml]: 1.0.3 is supported

**Comment by alexbocharov** [comment]
@sebastienros  Ok

**Comment by alexbocharov** [comment]
Hmmmm...

**Travis:**
```
The command "dotnet build -c Release" exited with 0.
52.91s
$ dotnet test -c Release ./test/Orchard.Tests/Orchard.Tests.csproj
Build started, please wait...
Build completed.

Test run for /Users/travis/build/OrchardCMS/Orchard2/test/Orchard.Tests/bin/Release/netcoreapp1.0/Orchard.Tests.dll(.NETCoreApp,Version=v1.0)
Microsoft (R) Test Execution Command Line Tool Version 15.0.0.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
No test is available in /Users/travis/build/OrchardCMS/Orchard2/test/Orchard.Tests/bin/Release/netcoreapp1.0/Orchard.Tests.dll. Make sure that installed test discoverers & executors, platform & framework version settings are appropriate and try again.




The command "dotnet test -c Release ./test/Orchard.Tests/Orchard.Tests.csproj" exited with 0.

Done. Your build exited with 0.
/Users/travis/.travis/job_stages: line 150: shell_session_update: command not found
```

**Comment by sebastienros** [comment]
First issue is that it should target netcoreapp1.1. I'll do it once the current build is finished.

the addon is necessary to support csproj (what I have read somewhere).

File: .travis.yml [hunk 3]
```diff
@@ -1,26 +1,25 @@
...
-env:
-  global:
-    - DOTNET_SKIP_FIRST_TIME_EXPERIENCE: true
-    - DOTNET_CLI_TELEMETRY_OPTOUT: 1   
-    - TEMP: "$TRAVIS_BUILD_DIR/.temp"
-mono: none
 os:
   - linux
   - osx
 osx_image: xcode8.2
```
Comment (Jetski5822) [inline .travis.yml]: osx_image: xcode8.3

File: .travis.yml [hunk 4]
```diff
@@ -1,26 +1,25 @@
 language: csharp
-sudo: false
+sudo: required
 dist: trusty
-dotnet: 1.0.0
+mono: none
+dotnet: 1.0.3
```
Comment (Jetski5822) [inline .travis.yml]: not 1.0.4?

Comment (alexbocharov) [inline .travis.yml]: 1.0.4 is not supported

**Comment by sebastienros** [comment]
I think we are good now. The tests don't pass on macos for now but it's an issue in dotnet test and xunit. https://github.com/Microsoft/vstest/issues/578

So as they are run in Linux and Windows we should be covered, and I assume at some point it will be fixed for macos.

I am removing the TEMP variable as it's not necessary anymore, and then we can merge if you agree.

**Comment by alexbocharov** [comment]
```
Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.12
 OS Platform: Darwin
 RID:         osx.10.12-x64
 Base Path:   /usr/local/share/dotnet/sdk/1.0.3


61.67s
$ dotnet restore
  Restoring packages for /Users/travis/build/OrchardCMS/Orchard2/src/Orchard.Cms.Web/Orchard.Cms.Web.csproj...
  Failed to download package 'Microsoft.Extensions.Options.1.0.0' from 'https://api.nuget.org/v3-flatcontainer/microsoft.extensions.options/1.0.0/microsoft.extensions.options.1.0.0.nupkg'.
  Too many open files
  Failed to download package 'Microsoft.Extensions.FileProviders.Physical.1.0.0' from 'https://api.nuget.org/v3-flatcontainer/microsoft.extensions.fileproviders.physical/1.0.0/microsoft.extensions.fileproviders.physical.1.0.0.nupkg'.
  Too many open files
  Failed to download package 'System.Diagnostics.DiagnosticSource.4.0.0' from 'https://api.nuget.org/v3-flatcontainer/system.diagnostics.diagnosticsource/4.0.0/system.diagnostics.diagnosticsource.4.0.0.nupkg'.
  Too many open files
  Failed to download package 'System.Runtime.InteropServices.RuntimeInformation.4.0.0' from 'https://api.nuget.org/v3-flatcontainer/system.runtime.interopservices.runtimeinformation/4.0.0/system.runtime.interopservices.runtimeinformation.4.0.0.nupkg'.
  Too many open files
  Failed to download package 'Microsoft.Extensions.Options.1.0.0' from 'https://api.nuget.org/v3-flatcontainer/microsoft.extensions.options/1.0.0/microsoft.extensions.options.1.0.0.nupkg'.
  Too many open files
  Failed to download package 'System.Buffers.4.0.0' from 'https://api.nuget.org/v3-flatcontainer/system.buffers/4.0.0/system.buffers.4.0.0.nupkg'.
  Too many open files
/usr/local/share/dotnet/sdk/1.0.3/NuGet.targets(97,5): error : Failed to download package 'Microsoft.Extensions.Options.1.0.0' from 'https://api.nuget.org/v3-flatcontainer/microsoft.extensions.options/1.0.0/microsoft.extensions.options.1.0.0.nupkg'. [/Users/travis/build/OrchardCMS/Orchard2/Orchard.sln]
```

**Comment by sebastienros** [comment]
@alexbocharov I'll let you merge once you think it's good

**Pull request merged**